controllers/etcd_controller.go

```diff
@@ -956,40 +964,57 @@ func (r *EtcdReconciler) updateEtcdErrorStatus(etcd *druidv1alpha1.Etcd, lastErr
 	lastErrStr := fmt.Sprintf("%v", lastError)
 	etcd.Status.LastError = &lastErrStr
 	etcd.Status.ObservedGeneration = &etcd.Generation
+	etcd.Status.Ready = nil
```

Comment:
Why is it set to `nil` here? You explained that the `Ready` field shall show the status of the etcd statefulset. Maybe the statefulset is ready but simply the reconciliation failed? I think it's cleaner to also call `health.CheckStatefulSet` here and set `Ready` accordingly.